migration/transitioning.rst

```diff
@@ -0,0 +1,93 @@
+.. _transitioning_mt:
+
+Citus extends PostgreSQL with distributed functionality, but it is not a drop-in replacement that scales out all workloads. A performant Citus cluster involves thinking about the data model, tooling, and choice of SQL features used. Migrating an existing single-node PostgreSQL database to Citus sometimes requires adjusting the schema and queries for optimal performance.
+
+Migration tactics differ between the two main Citus use cases of multi-tenant applications and real-time analytics. The former is more clear cut so we'll begin there.
+
+Multi-tenant Data Model
+-----------------------
+
+Citus is well suited to hosting B2B multi-tenant application data. In this model application tenants share a Citus cluster and a schema and Citus keeps each tenant's data private. Physically Citus co-locates the rows of each tenant's tables onto a single shard. Queries for a tenant route to the node which contains its shard.
+
+This section will explore how to model for the multi-tenant scenario, including necessary adjustments to the schema and queries.
+
+Schema Migration
+~~~~~~~~~~~~~~~~
+
+Migrating to a multi-tenant architecture requires a few changes to ensure each tenant's queries are routed to a single node. Once queries are routed this way (aka "pushed down") they can be executed without concern for the rest of the cluster. These queries can use the full features of SQL, including joins and transactions, without running into the inherent limitations of a distributed system.
+
+Tables and queries must simply include enough information that Citus knows where to place and route them. The first step to include this information is to examine existing tables and mentally classify each as either "global", "per-tenant," or "reference."
+
+*Global* tables aren't specific to individual tenants. They hold information about the system as a whole (e.g. admin tables for a website). Generally global tables keep to themselves and the application will not contain queries that join them with per-tenant tables nor will foreign key constraints hold between the two.
+
+On the other hand some tables may already have a column suitable to identify which tenant owns which rows. These are the natural *per-tenant* tables. Others don't have such a column, yet are not global because they are combined in queries with per-tenant tables. We call these *reference* tables and our task will be to adjust them for the multi-tenant use case.
```

Comment:
This section probably needs some re-organizing. The recommendations for adjusting reference tables comes in the Example section when we talk about the Distributor table.